Task-1 Terraform Done!

sean-ias opened the pull request:
Good day dear Aliaksei, terraform task 1 is done! Also would be glad, if you could review my other last 3 open pull requests which were about Kubernetes home tasks, somehow they are still open : ). Thank you!

grommir commented on task1_terraform/main.tf:
```diff
@@ -0,0 +1,16 @@
+# include at least one special character when special var is set to true:
+locals {
+  min_special = var.special ? 1 : 0
```
А что если нам потребуется другое количество специальных символов?

sean-ias replied:
Чтобы изменить количество спец. символов, нам потребуется изменить часть вот этого кода _min_special = var.special ? **x** : 0_ где например для минимум 2 спец. символов, поставим цифро 2 на место **x**. 
Выглядет вот так:
_locals {
  min_special = var.special ? 2 : 0
  }_

grommir commented on task1_terraform/main.tf:
```diff
@@ -0,0 +1,16 @@
+  length  = var.length
+  special = var.special
+  override_special = var.override_special
+  min_lower = 1
+  min_upper = 1
+  min_numeric = 1
```
Почему тут мы используем захардкоданые значения вместо переменных?

sean-ias replied:
Здесь я решил захардкодить эти значение чтобы убедиться что в каждем пароле было минимум одна маленькая, большая буква и цифра. Без этого раньше когда по дефолту стоял min_lower=0, я получил несколько кейсов где не было вообще маленьких букв, только были большие буквы и цифры с спец. символами. Это конечно, я подумал, будет влиять на сложность парола!

grommir commented:
Я не вижу паролей в outputs
![image](https://user-images.githubusercontent.com/25118604/235101996-723e2248-de21-4081-b5a3-c861849a5056.png)

sean-ias commented:
Fixed output sensitivity issue, now we have the output of passwords' list.
Also added sensitive config, and tfstate files into .gitignore files.
Thank you!
